### PR TITLE
Only update and purge the summary if it actually changed

### PR DESCRIPTION
**Pull request by Pchelolo**
Bug: https://phabricator.wikimedia.org/T163924
cc @wikimedia/services

### CODE REVIEW

File: sys/key_value.js [hunk 1]
```diff
@@ -194,6 +195,32 @@ class KVBucket {
...
             return { status: 400 };
         });
+
+        if (req.query.ignore_duplicates) {
+            return hyper.get({
+                uri: new URI([rp.domain, 'sys', 'key_value', rp.bucket, rp.key])
+            })
+            .then((oldContent) => {
+                if (stringify(req.body) === stringify(oldContent.body) &&
+                    (!req.headers['content-type']
```
Comment (d00rman) [inline sys/key_value.js]: Nit: let's indent this one more level so as not to confuse it with the statements inside the `if` block

File: sys/key_value.js [hunk 2]
```diff
@@ -194,6 +195,32 @@ class KVBucket {
...
+                        }
+                    };
+                } else {
+                    throw new HTTPError({
+                        status: 404
+                    });
+                }
+            })
+            .catch({ status: 404 }, doPut);
+        } else {
```
Comment (d00rman) [inline sys/key_value.js]: Nit: no need for `else` here, as there is a `return` inside the `if`

**Comment by d00rman** [comment]
Some nits, but otherwise good work! :+1:

**Comment by Pchelolo** [comment]
@d00rman Updated!

File: sys/key_value.js [hunk 3]
```diff
@@ -194,6 +195,29 @@ class KVBucket {
...
+            return hyper.get({
+                uri: new URI([rp.domain, 'sys', 'key_value', rp.bucket, rp.key])
+            })
+            .then((oldContent) => {
+                if (stringify(req.body) === stringify(oldContent.body) &&
+                        (!req.headers['content-type']
+                        || req.headers['content-type'] === oldContent.headers['content-type'])) {
+                    hyper.metrics.increment(`sys_kv_${req.params.bucket}.unchanged_rev_render`);
+                    return {
+                        status: 200,
```
Comment (gwicke) [inline sys/key_value.js]: According to https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6, 204 (no content) might be more appropriate here.

Comment (Pchelolo) [inline sys/key_value.js]: @gwicke Done!

File: v1/definition.yaml [hunk 4]
```diff
@@ -123,7 +123,18 @@ paths:
             request:
               method: put
               uri: /{domain}/sys/key_value/term.definition/{request.params.term}
-              headers: '{{extract.headers}}'
+              headers: '{{merge({"if-none-hash-match": "*"}, extract.headers)}}'
+              body: '{{extract.body}}'
+            # With the ignore_duplicates parameter the storage will return 200 if the content is not changed,
```
Comment (d00rman) [inline v1/definition.yaml]: *With the if-none-hash-match header the storage will return 204*...

File: v1/summary.yaml [hunk 5]
```diff
@@ -139,21 +139,30 @@ paths:
...
                 description: '{{extractDescription(extract.body.items[0].terms)}}'
                 coordinates: '{{processCoords(extract.body.items[0].coordinates)}}'
-
+        - store_and_return:
+            request:
+              method: put
+              uri: /{domain}/sys/key_value/summary/{request.params.title}
+              headers: '{{merge({"if-none-hash-match": "*"}, extract.headers)}}'
+              body: '{{extract.body}}'
+            # With the ignore_duplicates parameter the storage will return 200 if the content is not changed,
```
Comment (d00rman) [inline v1/summary.yaml]: Idem as above

File: v1/definition.yaml (hunk 4 above)
Comment (Pchelolo) [inline v1/definition.yaml]: Errr.... Fixed.

**Pull request merged**